tools/utils/categories.ts

```diff
@@ -25,12 +25,8 @@ for (const categoryId of categoryIds) {
 
   for (const rule of rules) {
     const ruleCategories = rule.meta.docs.categories
-    // Throw if rule does not have a category
-    if (!ruleCategories?.length) {
-      throw new Error(`Rule "${rule.ruleId}" does not have any category.`)
-    }
 
-    if (ruleCategories.includes(categoryId)) {
+    if (ruleCategories?.includes(categoryId)) {
       categoriesConfig[categoryId].rules?.push(rule)
     }
   }
```

Comment:
This part is no longer needed now as there is a mechanism to ignore rules without categories, and I've been able to remove it entirely on my rebased branch.